pkg/controller/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -636,6 +582,169 @@ func TestToolchainStatusConditions(t *testing.T) {
+				LastTransitionTime: metav1.Time{time.Now()},
+			},
+		}
+		defer func() { toolchainStatus.Status.Conditions = nil }()
+
+		reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(),
+			newGetMemberClustersFuncReady, hostOperatorDeployment, memberStatus, registrationServiceDeployment,
+			registrationService, toolchainStatus)
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, requeueResult, res)
+
+		AssertThatToolchainStatus(t, req.Namespace, requestName, fakeClient).
+			HasCondition(componentsReady()).
+			HasHostOperatorStatus(hostOperatorStatusReady()).
+			HasMemberStatus(memberClusterSingleReady()).
+			HasRegistrationServiceStatus(registrationServiceReady())
+
+		// Lookup the notification
+		var notification toolchainv1alpha1.Notification
+		err = fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs, "toolchainstatus-unready"),
+			&notification)
+		require.Error(t, err)
+		require.IsType(t, &errors.StatusError{}, err)
+		require.Equal(t, string(err.(*errors.StatusError).ErrStatus.Reason), "NotFound")
+	})
+
+	t.Run("Notification not created when host operator deployment not ready within threshold", func(t *testing.T) {
+		// given
+		defer counter.Reset()
+		hostOperatorDeployment := newDeploymentWithConditions(defaultHostOperatorName,
+			status.DeploymentNotAvailableCondition(), status.DeploymentProgressingCondition())
+
+		os.Setenv("HOST_OPERATOR_CONFIG_MAP_NAME", "notification_test_config")
+		os.Setenv("WATCH_NAMESPACE", test.HostOperatorNs)
+		config := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "notification_test_config",
+				Namespace: test.HostOperatorNs,
+			},
+			Data: map[string]string{
+				"admin.email": "admin@dev.sandbox.com",
+			},
+		}
+
+		toolchainStatus.Status.Conditions = []toolchainv1alpha1.Condition{
+			{
+				Type:               toolchainv1alpha1.ConditionReady,
+				Status:             corev1.ConditionFalse,
+				LastTransitionTime: metav1.Time{time.Now()},
+				Reason:             toolchainv1alpha1.ToolchainStatusComponentsNotReadyReason,
+			},
+		}
+		defer func() { toolchainStatus.Status.Conditions = nil }()
+
+		reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(),
+			newGetMemberClustersFuncReady, hostOperatorDeployment, memberStatus, registrationServiceDeployment,
+			registrationService, toolchainStatus, config)
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, requeueResult, res)
+
+		// Lookup the notification
+		var notification toolchainv1alpha1.Notification
+		err = fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs, "toolchainstatus-unready"),
+			&notification)
+		require.Error(t, err)
+		require.IsType(t, &errors.StatusError{}, err)
+		require.Equal(t, string(err.(*errors.StatusError).ErrStatus.Reason), "NotFound")
```

Comment:
It's a duplicate assertion from a previous test. Can we move it to `AssertToolchainStatusNotificationNotCreated()` function so we can reuse it everywhere?